core/sdk-core/src/main/java/software/amazon/awssdk/core/checksums/Algorithm.java

```diff
@@ -44,12 +47,16 @@ public static Algorithm fromValue(String value) {
             return null;
         }
         String normalizedValue = StringUtils.lowerCase(value);
-        return Arrays.stream(values())
-                .filter(algorithm -> algorithm.value.equals(normalizedValue))
-                .findFirst()
-                .orElseThrow(() -> new IllegalArgumentException(String.format("Unknown Algorithm '%s'", normalizedValue)));
+        if (!VALUE_MAP.containsKey(normalizedValue)) {
+            throw new IllegalArgumentException("The provided value is not a valid algorithm " + value);
+        }
+        return VALUE_MAP.get(normalizedValue);
```

Comment:
Since we're being picky about performance, should we just do a get and check to see if the result is null? (I don't think we need null to be a valid algorithm)

Reply:
Yeah, updated.